File: backend/domain/training_session.py
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class TrainingSession:
    """表示一次训练会话，关联了模型和使用的语料库。"""

    id: str
    name: str
    base_model: str
    start_time: datetime
    last_trained: datetime
    metrics: dict = field(default_factory=dict)
    tokens_trained: int = 0

    def update_metrics(self, new_metrics: dict):
        self.metrics.update(new_metrics)
        self.last_trained = datetime.now()

    def increment_tokens_trained(self, tokens: int):
        self.tokens_trained += tokens

File: app/lib/models/training_session.dart
class TrainingSession {
  final String id;
  final String name;
  final String baseModel;
  final DateTime startTime;
  final DateTime lastTrained;
  final DateTime? endTime;
  final int tokensTrained;
  final Map<String, dynamic> metrics;

  TrainingSession({
    required this.id,
    required this.name,
    required this.baseModel,
    required this.startTime,
    required this.lastTrained,
    this.endTime,
    required this.tokensTrained,
    required this.metrics,
  });

  factory TrainingSession.fromJson(Map<String, dynamic> json) {
    return TrainingSession(
      id: json['id'],
      name: json['name'],
      baseModel: json['base_model'],
      startTime: DateTime.parse(json['start_time']),
      lastTrained: DateTime.parse(json['last_trained']),
      endTime:
          json['end_time'] != null ? DateTime.parse(json['end_time']) : null,
      tokensTrained: json['tokens_trained'] ?? 0,
      metrics: json['metrics'],
    );
  }
}
